server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -266,6 +271,121 @@ public void testExtensionsInitialize() throws Exception {
+
+        List<String> extensionsYmlLines = Arrays.asList(
+            "extensions:",
+            "   - name: firstExtension",
+            "     uniqueId: uniqueid1",
+            "     hostName: 'myIndependentPluginHost1'",
+            "     hostAddress: '127.0.0.0'",
+            "     port: '9300'",
+            "     version: '3.0.0'",
+            "   - name: secondExtension",
+            "     uniqueId: 'uniqueid2'",
+            "     hostName: 'myIndependentPluginHost2'",
+            "     hostAddress: '127.0.0.1'",
+            "     port: '9301'",
+            "     version: '2.0.0'"
+        );
+        Files.write(extensionDir.resolve("extensions.yml"), extensionsYmlLines, StandardCharsets.UTF_8);
+
+        Path pluginDir1 = extensionDir.resolve("fake-extension-1");
+
+        PluginTestUtil.writePluginProperties(
+            pluginDir1,
+            "description",
+            "fake desc",
+            "name",
+            "firstExtension",
+            "version",
+            "1.0",
+            "opensearch.version",
+            Version.CURRENT.toString(),
+            "java.version",
+            System.getProperty("java.specification.version"),
+            "classname",
+            "FakeExtensions"
+        );
+
+        final ClusterService clusterService = mock(ClusterService.class);
+        Settings settings = Settings.builder().put("node.name", "my-node").build();
+        ClusterName clusterName = ClusterName.CLUSTER_NAME_SETTING.get(settings);
+        ThreadPool threadPool = new TestThreadPool(ExtensionsOrchestratorTests.class.getSimpleName());
+        MockNioTransport transport = new MockNioTransport(
+            settings,
+            Version.CURRENT,
+            threadPool,
+            new NetworkService(Collections.emptyList()),
+            PageCacheRecycler.NON_RECYCLING_INSTANCE,
+            new NamedWriteableRegistry(Collections.emptyList()),
+            new NoneCircuitBreakerService()
+        );
+        TransportService transportService = new MockTransportService(
+            settings,
+            transport,
+            threadPool,
+            TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+            (boundAddress) -> new DiscoveryNode(
+                "test_node",
+                "test_node",
+                boundAddress.publishAddress(),
+                emptyMap(),
+                emptySet(),
+                Version.CURRENT
+            ),
+            null,
+            Collections.emptySet()
+        );
+
+        ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
+
+        transportService.registerRequestHandler(
+            ExtensionsOrchestrator.REQUEST_EXTENSION_CLUSTER_STATE,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            (request, channel, task) -> {
+                channel.sendResponse(new ClusterStateResponse(clusterName, null, false));
```

Comment:
Why are we creating a new `ClusterStateResponse` here? 
Ideally we should be able to test the response coming from `extensionsOrchestrator.handleExtensionRequest(request)`

Or Am I missing something here?

Reply:
Added another test [here](https://github.com/opensearch-project/OpenSearch/pull/3750/files#diff-4e10e710ab4dc067ee8aa4dccb15f340690908cd846d3c8124dc3b17595b9ab4R334) to test the response. 
This test focusses on mocking `registerRequestHandler`